README.md

```diff
@@ -0,0 +1,194 @@
+    return async (dispatch) => {
+        dispatch({ type: 'FETCH_ORDERS_START' });
+
+        try {
+            const orders = await fetch('/account/orders');
+
+            dispatch({ type: 'FETCH_ORDERS_SUCCESS', payload: orders });
+        } catch (err) {
+            dispatch({ type: 'FETCH_ORDERS_FAIL', payload: err });
+            throw err;
+        }
+    }
+}
+
+store.dispatch(login('my-username', '12345'));
+
+expect(store.getActions()).toContain([
+    'LOGIN_START',
+    'FETCH_ORDERS_SUCCESS'
+]);
+```
+
+The assertion above will fail because `FETCH_ORDERS_SUCCESS` will not yet exist in the stack of actions.
+To solve this, one can use `setTimeout` explicitly in each test:
+
+```js
+store.dispatch(login('my-username', '12345'));
+
+setTimeout(() => expect(store.getActions()).toContain([
+    'LOGIN_START',
+    'FETCH_ORDERS_SUCCESS'
+]), 50);
+```
+
+However, this is not pretty and is error-prone. `redux-mock-store-await-actions` makes this easier for you.
+
+## Usage
+
+### Example #1: action types
+
+Supply the action types to await for.
+
+```js
+import configureStore from 'redux-mock-store';
+import thunkMiddleware from 'redux-thunk';
+
+const store = configureStore([thunkMiddleware])();
+
+store.dispatch(login('my-username', '12345'));
+await waitForActions(store, ['LOGIN_START', 'FETCH_ORDERS_SUCCESS']);
+```
+
+### Example #2: action objects
+
+Supply the action objects to await for. Matches a subset of the properties of the dispatched actions.
+
+```js
+import configureStore from 'redux-mock-store';
+import thunkMiddleware from 'redux-thunk';
+
+const store = configureStore([thunkMiddleware])();
+
+store.dispatch(login('my-username', '123'));
+await waitForActions(store, [
+    {
+        type: 'LOGIN_START',
+        payload: { username: 'my-username' },
+    },
+    {
+        type: 'FETCH_ORDERS_SUCCESS',
+    },
+]);
+```
+
+### Example #3: function (advanced use cases)
+
+Supply a predicate which will be called for every action dispatched.
+
+```js
+import configureStore from 'redux-mock-store';
+import thunkMiddleware from 'redux-thunk';
+
+const store = configureStore([thunkMiddleware])();
+
+store.dispatch(login('my-username', '123'));
+await waitForActions(store, (storeActions) => {
+	const hasLoginStart = storeActions.some((item) => item.type === 'LOGIN_START' && item.payload.username === 'my-username');
+    const hasFetchOrdersSuccess = storeActions.some((item) => item.type === 'FETCH_ORDERS_SUCCESS');
+
+    return hasLoginStart && hasFetchOrdersSuccess;
+});
+```
+
+**NOTE:** Subsequent calls to `waitForActions` should be preceded by a call to `store.clearActions()`, otherwise the returned `Promise` will resolve immediately.
+
+## API
+
+### waitForActions(store, actions, [timeout])
+
+Returns a `Promise` which fulfils if all `actions` are dispatched before the timeout expires, otherwise it is rejected. The `Promise` has a `.cancel()` function which, if called, will reject the `Promise`.
```

Comment:
fulfils -> fulfulls (prefer American english)

Reply:
fulfills you mean

Reply:
Yeap, fixed.